x/bank/keeper/keeper.go

```diff
@@ -30,6 +31,9 @@ type Keeper interface {
 	GetSupply(ctx sdk.Context) exported.SupplyI
 	SetSupply(ctx sdk.Context, supply exported.SupplyI)
 
+	GetDenomMetaData(ctx sdk.Context) []types.Metadata
+	SetDenomMetaData(ctx sdk.Context, denomMetaData []types.Metadata)
```

Comment:
I think we should get and set this metadata per denom not as one blob.

Reply:
Sometimes we do need to retrieve all the denomination metadata from the store as in this case:

![image](https://user-images.githubusercontent.com/30392990/88438319-be8a0d00-ce08-11ea-95c0-72bf8182a15a.png)

**so can we have the keeper interface as follow**?
```go
...
        GetAllDenomMetadata(ctx sdk.Context) []types.Metadata
	GetDenomMetaData(ctx sdk.Context, denom string) types.Metadata
	SetDenomMetaData(ctx sdk.Context, denomMetaData types.Metadata)
...
```

Reply:
How about we just have a method to iterate through everything rather than returning a single blob?

Reply:
Agree that we should opt to provide an iterator method that can be used to get all the resources instead of a `GetAll*` method. This is consistent with other modules.